### PR TITLE
Disable source fortification

### PR DESCRIPTION
**Pull request by adombeck**
Using the _FORTIFY_SOURCE macro together with ASan is currently not supported:

    https://github.com/google/sanitizers/issues/247

### CODE REVIEW

**Review by fmeum — APPROVED** [review]

**Comment by adombeck** [comment]
Now also disables source fortification when building for coverage and when using other build systems

File: internal/build/other/other.go [hunk 1]
```diff
@@ -232,6 +235,9 @@ func (b *Builder) setCoverageEnv() error {
 		// ----- Flags used to build with code coverage -----
 		"-fprofile-instr-generate",
 		"-fcoverage-mapping",
+		// Disable source fortification to ensure that coverage builds
+		// reach the same code as ASan builds.
```
[the remark this answers is not in the document]
Comment (adombeck) [reply internal/build/other/other.go:239]: How can I make it more clear? Like this maybe?

```
// Same as for ASan builds, we disable source fortification for 
// coverage builds, to ensure that both builds reach the same code.
```

[the remark this answers is not in the document]
Comment (fmeum) [reply internal/build/other/other.go:239]: The problem is that they won't reach the same code, right? Without fortification and sanitizers, coverage builds will progress through the location of the memory issue, potentially (non-deterministically) executing more code than any fuzzing build.

[the remark this answers is not in the document]
Comment (fmeum) [reply internal/build/other/other.go:239]: Sounds good!

[the remark this answers is not in the document]
Comment (adombeck) [reply internal/build/other/other.go]: done

**Pull request merged**